core/src/main/java/org/eobjects/datacleaner/extension/output/CreateCsvFileAnalyzer.java

```diff
@@ -107,16 +127,94 @@ public OutputWriter createOutputWriter() {
         for (int i = 0; i < headers.length; i++) {
             headers[i] = columns[i].getName();
         }
-        return CsvOutputWriterFactory.getWriter(file.getPath(), headers, separatorChar, quoteChar, escapeChar,
-                includeHeader, columns);
+        return CsvOutputWriterFactory.getWriter(_tempFile.getPath(), headers, separatorChar, quoteChar, escapeChar, includeHeader, columns);
     }
 
     @Override
     protected WriteDataResult getResultInternal(int rowCount) {
+        final CsvConfiguration csvConfiguration = new CsvConfiguration(CsvConfiguration.DEFAULT_COLUMN_NAME_LINE, FileHelper.DEFAULT_ENCODING,
+                separatorChar, quoteChar, escapeChar, false, true);
+        final CsvDataContext tempDataContext = new CsvDataContext(_tempFile, csvConfiguration);
+        final Table table = tempDataContext.getDefaultSchema().getTable(0);
+
+        final Comparator<? super Row> comparator = new Comparator<Row>() {
+            @SuppressWarnings("unchecked")
+            @Override
+            public int compare(Row row1, Row row2) {
+                Comparable<Object> value1 = (Comparable<Object>) row1.getValue(columnToBeSortedOn.getPhysicalColumn().getColumnNumber());
+                Comparable<Object> value2 = (Comparable<Object>) row2.getValue(columnToBeSortedOn.getPhysicalColumn().getColumnNumber());
```

Comment:
calling Row.getValue with the columnToBeSortedOn is not going to work... Maybe it has worked by coincidence, but it's certainly not a given. The reason is that the rows may not even contain this column (that's dependent on the @Configured InputColumn[] that drives the inserts) and it's index in the file may not be the same as it's original column number.

I'd suggest figuring out what the index is at the time of inserting the records into the temp file. Maybe first check if columnToBeSortedOn is included in the array of InputColumns that are being written. If not, it needs to be added somehow. If it is, then keep it's index so that it can be used here.